src/sentry/api/event_search.py

```diff
@@ -1613,75 +1668,83 @@ def validate_argument_count(self, field, arguments):
             "percentile",
             required_args=[NumericColumnNoLookup("column"), NumberRange("percentile", 0, 1)],
             aggregate=[u"quantile({percentile:g})", ArgValue("column"), None],
-            result_type="duration",
+            result_type_fn=lambda args, columns: args[0].get_type(columns[0]),
```

Comment:
Could we move this to a function, something like `reflective_result_type(0)` perhaps?